timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceImpl.java

```diff
@@ -257,24 +260,22 @@ public ListenableFuture<TimestampRange> getFreshTimestampsAsync(int timestampsTo
     }
 
     @Override
-    public ListenableFuture<TimestampLeaseResponse> acquireTimestampLease(
-            TimestampLeaseName timestampName, UUID requestId, int numFreshTimestamps) {
+    public ListenableFuture<TimestampLeaseResponses> acquireTimestampLease(
+            UUID requestId, Map<TimestampLeaseName, Integer> numFreshTimestamps) {
         long timestamp = timestampService.getFreshTimestamp();
```

Comment:
They all use the same freshTs to lock. This should not matter - the different timestamp lease names define orthogonal namespaces.

I can easily change this.

Reply:
I think that's right, min leases are always allowed to go backwards